app/views/articles/index.html.haml

```diff
@@ -1,18 +1,29 @@
 %h1 List of articles
+= render 'search', q: @q
```

Comment:
major use case of partials is to make the code dry. Not sure if its applicable here.